src/NUnitFramework/framework/Internal/ParamNumberConversions.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections;
+using System.Reflection;
+using NUnit.Compatibility;
+using NUnit.Framework.Interfaces;
+
+namespace NUnit.Framework.Internal
+{
+    /// <summary>
+    /// Helper methods for converting parameters to numeric values
+    /// </summary>
+    internal static class ParamNumberConversions
```

Comment:
Namespace is good, glad to see this is internal aswell!

Reply:
One concern, this class doesn't just comvert numbers now we've also included the functionality from ValuesAttribute! Perhaps `ParamAttributeTypeConversions` would be a better name, or something similar? A few of the comments in this class also specify 'numeric', when the methods convert other types too.

Sorry - I think this was the name I suggested - at the time, we were just talking about it having the functionality from RangeAttribute!